### DISCUSSION TITLE
I need some help with dymanic content

### DISCUSSION THREAD

**Discussion by elfeffe** [livewire/livewire#1114] [4. General]
I need to show a dropdown where the user selects a type.
When he selects that type, I need to update another dropdown with options based on that type.
Ok, I have found this tutorial http://novate.co.uk/dynamic-cascading-dropdown-with-livewire/

But my problem is that I don't know if there will be a type or not, in some cases, I just have 1 dropdown, but in some other, I have 2.

So I can't understand if I can do that with LW.
I have a form, and inside I render LW components, N depending on how many I need. When the user selects, I emit the value and the name of that select, and I plan to catch that with the form (the master component) to refresh the rest.

But I'm not getting how to do that, how to render and control dynamic components inside my form.
All the examples that I find are based on the fact that you know the models, but how to deal with dynamic selects inside your form?

How to let the form know that your first select has a value, and that you have another one that needs to be refreshed with new data based on the first select?

To show an example, I want to do something like Amazon does:
https://www.amazon.com/ASICS-Gel-Venture-Running-Shoes-Black/dp/B07JH5L53L/ref=sr_1_1?dchild=1&keywords=asics&qid=1592412523&sr=8-1

Do you see the dropdown? When you choose an option, it refreshes the rest.
I plan to have the attributes on my DB, so in some cases, I just have one option to choose, but in some others, I have 2 like in the amazon example.
The thing is that this is dynamic, and I don't know how many child selects can the be inside my component.

**Comment by nuernbergerA** [comment]
You could simply use livewire's amazing event system and [fire an event](https://laravel-livewire.com/docs/events#firing-events).

Sticking to the shop example and the possibility to have different use cases with more or less select options.

- Given we have an Article **shoe** with a select for **size** `[7,8,9,10]`
- Each size would translate to an SKU/article number.
- On the first render, I would set the "default" SKU
- And if the size would change, it will fire an event with the corresponding SKU

With this pattern, you can have non to "infinite" select options.

Cheers
